docs/guide/cache/fastly.md

```diff
@@ -0,0 +1,464 @@
+Updated at: 2022-06-23 12:24:48 +0000 UTC
+
+```
+
+You may also get the same details for a particular snippet using the `vcl snippet describe` command :
+
+```
+fastly vcl snippet describe --version=active "--name=Re-Enable shielding on restart"
+```
+
+### Create a snippet
+
+```
+fastly vcl snippet create --name="Re-Enable shielding on restart" --version=active --autoclone --priority 100 --type recv --content=vendor/ezsystems/ezplatform-http-cache-fastly/fastly/snippet_re_enable_shielding.vcl
+fastly service-version activate --version=latest
+```
+
+### Delete a snippet
+
+```
+fastly vcl snippet delete --name="Re-Enable shielding on restart" --version=active --autoclone
+fastly service-version activate --version=latest
+```
+
+### Update an existing snippet
+
+```
+fastly vcl snippet update --name="Re-Enable shielding on restart" --version=active --autoclone --priority 100 --type recv --content=vendor/ezsystems/ezplatform-http-cache-fastly/fastly/snippet_re_enable_shielding.vcl
+fastly service-version activate --version=latest
+```
+
+### Get the diff between two versions
+
+Using the Fastly web interface you can easily see the diff between two different versions. Unfortunately, this is not supported by the Fastly CLI. However, it is possible with the help of the Fastly API and GNU diff.
+
+You use the Fastly API to download the generated VCL. This generated VCL is the vcl config that Fastly generates based on all the configuration settings ( so it includes all custom .vcls, snippets, origin config, just everything).
+
+The example below extracts the generated vcl for version no. 11 of some service
+```
+curl -i  "https://api.fastly.com/service/[FASTLY_SERVICE_ID]/version/11/generated_vcl" -H "Fastly-Key: [FASTLY_API_TOKEN]" -H "Accept: application/json" > generated_vcl_11_raw
+cp generated_vcl_11_raw generated_vcl_11_json_only
+```
+
+Next, you need to edit generated_vcl_11_json_only in your favourite editor, remove anything before the json data and save.
+Then, do the same steps again for version no. 12 (or whatever version you want to diff version 11 against)
+
+Then replace \n in the files in order to give human-readable diffs:
+
+```
+cat generated_vcl_11_json_only |jq .content|perl -pe 's/\\n/\n/g' > generated_vcl_11_json_done
+cat generated_vcl_12_json_only |jq .content|perl -pe 's/\\n/\n/g' > generated_vcl_12_json_done
+```
+
+Finally, you can use GNU diff to and get a readable diff of the two versions:
+
+```
+diff -ruN generated_vcl_11_json_done generated_vcl_12_json_done
+```
+
+## Enabling Basic-Auth on Fastly.
+
+In order to enable basic-auth, you can use https://developer.fastly.com/solutions/examples/http-basic-auth as an example.
+
+Here is a step-by-step guide on how to make that work.
+
+Usernames and password can be stored inside vcl, but we'll instead store them in a [dictionary](https://docs.fastly.com/en/guides/working-with-dictionaries-using-the-web-interface#working-with-dictionaries-using-vcl-snippets)
```

Comment:
```suggestion
Usernames and passwords can be stored inside vcl, but in these case it will be stored in a [dictionary](https://docs.fastly.com/en/guides/working-with-dictionaries-using-the-web-interface#working-with-dictionaries-using-vcl-snippets)
```

Reply:
@juskora : not "this case" ?

Reply:
ping @juskora

Reply:
This case, yes you're right, sorry my bad.